server/src/main/java/com/microsoft/java/bs/core/internal/managers/BuildTargetManager.java

```diff
@@ -82,13 +83,11 @@ public List<BuildTargetIdentifier> store(GradleSourceSets gradleSourceSets) {
         changedTargets.add(btId);
       }
       newCache.put(btId, buildTarget);
-      // Store the relationship between the project dir and the build target id.
-      // 'test' and other source sets are ignored.
-      if ("main".equals(sourceSet.getSourceSetName())) {
-        projectPathToBuildTargetId.put(sourceSet.getProjectDir().getAbsolutePath(), btId);
-      }
```

Comment:
@jdneo The issue with project dependencies exception might be down to me removing this.  Now all sourceset dependencies are returned which I think is correct.
E.g. `IntegTests` might depend on `testFixtures` and `main`

Reply:
Ok, never mind. I've already add dedup logic at client side: https://github.com/microsoft/vscode-gradle/pull/1568

This is more likely a client side issue due to the limited capability of client.